### PR TITLE
feat(api): add API to open/close cluster

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

Add API to open/close cluster

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @supereagle @cd1989 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/server/handler/v1alpha1/integration.go [hunk 1]
```diff
@@ -484,6 +492,101 @@ func DeleteIntegration(ctx context.Context, tenant, name string) error {
 	return cerr.ConvertK8sError(err)
 }
 
+// OpenCluster opens cluster type integration to execute workflow
+func OpenCluster(ctx context.Context, tenant, name string) error {
+	isName := svrcommon.IntegrationSecret(name)
```
Comment (cd1989) [inline pkg/server/handler/v1alpha1/integration.go]: `isName` hard to understand

File: pkg/server/handler/v1alpha1/integration.go [hunk 2]
```diff
@@ -444,6 +445,13 @@ func DeleteIntegration(ctx context.Context, tenant, name string) error {
 		return err
 	}
 
+	if in.Spec.Type == api.Cluster && in.Spec.Cluster != nil {
+		cluster := in.Spec.Cluster
+		if cluster.IsWorkerCluster {
```
Comment (cd1989) [inline pkg/server/handler/v1alpha1/integration.go]:
```
if in.Spec.Type == api.Cluster && in.Spec.Cluster != nil  && in.Spec.Cluster.IsWorkerCluster {
}
```

File: pkg/server/handler/v1alpha1/integration.go [hunk 3]
```diff
@@ -484,6 +492,101 @@ func DeleteIntegration(ctx context.Context, tenant, name string) error {
...
+	}
+
+	in.Spec.Cluster.IsWorkerCluster = true
+	secret, err := buildSecret(tenant, in)
+	if err != nil {
+		return err
+	}
+
+	ns := svrcommon.TenantNamespace(tenant)
+	_, err = handler.K8sClient.CoreV1().Secrets(ns).Update(secret)
```
Comment (cd1989) [inline pkg/server/handler/v1alpha1/integration.go]: retryOnConflict

File: pkg/server/handler/v1alpha1/integration.go [hunk 4]
```diff
@@ -484,6 +492,101 @@ func DeleteIntegration(ctx context.Context, tenant, name string) error {
...
+	// close cluster for the tenant, delete namespace
+	err = CloseClusterForTenant(in, tenant)
+	if err != nil {
+		return err
+	}
+
+	in.Spec.Cluster.IsWorkerCluster = false
+
+	if cluster.Namespace == svrcommon.TenantNamespace(tenant) {
+		in.Spec.Cluster.Namespace = ""
```
Comment (cd1989) [inline pkg/server/handler/v1alpha1/integration.go]: Status change shouldn't change `Spec`

File: pkg/server/handler/v1alpha1/integration.go [hunk 5]
```diff
@@ -484,6 +492,101 @@ func DeleteIntegration(ctx context.Context, tenant, name string) error {
...
+		in.Spec.Cluster.PVC = ""
+	}
+
+	secret, err := buildSecret(tenant, in)
+	if err != nil {
+		return err
+	}
+
+	ns := svrcommon.TenantNamespace(tenant)
+	_, err = handler.K8sClient.CoreV1().Secrets(ns).Update(secret)
```
Comment (cd1989) [inline pkg/server/handler/v1alpha1/integration.go]: retryOnConflict

File: pkg/server/handler/v1alpha1/integration.go (hunk 2 above)
Comment (zhujian7) [reply pkg/server/handler/v1alpha1/integration.go]: ok

**Comment by zhujian7** [comment]
@cd1989 All comments have been fixed, PTAL

**Pull request merged**